pkg/globalnet/controllers/uninstall.go

```diff
@@ -216,3 +218,32 @@ func RemoveGlobalIPAnnotationOnNode(cfg *rest.Config) {
 
 	logger.Infof("Successfully removed globalIP annotation from node %q", nodeName)
 }
+
+func RemoveStaleInternalServices(config *syncer.ResourceSyncerConfig) error {
+	_, gvr, err := util.ToUnstructuredResource(&corev1.Service{}, config.RestMapper)
+	if err != nil {
+		return errors.Wrap(err, "error converting resource")
+	}
+
+	services := config.SourceClient.Resource(*gvr)
+
+	svcList, err := services.Namespace(corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error listing the services")
+	}
+
+	for i := range svcList.Items {
+		svc := &svcList.Items[i]
+		if !svc.GetDeletionTimestamp().IsZero() {
+			if finalizer.IsPresent(svc, InternalServiceFinalizer) {
```

Comment:
If we search by the label as I suggested above then we don't really need to check for the `InternalServiceFinalizer`, ie it should be present.